services/worker/src/worker/job_runners/_job_runner_with_datasets_cache.py

```diff
@@ -28,14 +29,18 @@ def __init__(
         )
 
     def set_datasets_cache(self, cache_subdirectory: Optional[Path]) -> None:
-        datasets.config.HF_DATASETS_CACHE = cache_subdirectory
+        if cache_subdirectory is None:
+            return
+        datasets.config.HF_DATASETS_CACHE = cache_subdirectory / "datasets"
         logging.debug(f"datasets data cache set to: {datasets.config.HF_DATASETS_CACHE}")
         datasets.config.DOWNLOADED_DATASETS_PATH = (
             datasets.config.HF_DATASETS_CACHE / datasets.config.DOWNLOADED_DATASETS_DIR
         )
         datasets.config.EXTRACTED_DATASETS_PATH = (
             datasets.config.HF_DATASETS_CACHE / datasets.config.EXTRACTED_DATASETS_DIR
         )
+        huggingface_hub.constants.HF_HUB_CACHE = cache_subdirectory / "hub"
```

Comment:
cc @Wauplin please keep this working in the future ^^'

Reply:
Any chance you could use the env variables instead of monkeypatching?